### DISCUSSION TITLE
issue: secp256k1.privateKeyVerify is not a function

### DISCUSSION THREAD

**Discussion by farairo** [ConsenSys-archive/truffle#6222] [General Q&A]
code: new HDWalletProvider(walletKey, providerUrl);
I am getting following error with truffle 
secp256k1.privateKeyVerify is not a function at Object.exports.privateKeyVerify (C:\Users\user\OneDrive\Desktop\bot\node_modules\@truffle\hdwallet-provider\node_modules\ethereumjs-util\dist\secp256k1v3-adapter.js

any idea how to fix it?

**Comment by MadelineMurray** [comment]
hi @farairo  - Truffle is being sunset. For more information and migration options, click [here](https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat?utm_source=github&utm_medium=referral&utm_campaign=2023_Sep_truffle-sunset-2023_announcement_).
